tests/pxScene2d/test_imagecache.cpp

```diff
@@ -0,0 +1,465 @@
+#include "gtest/gtest.h"
+#include <list>
+#define private public
+#define protected public
+#include "pxFileCache.h"
+#include "pxHttpCache.h"
+#include "pxResource.h"
+#include "rtString.h"
+
+class commonTestFns
+{
+  protected:
+     rtError addDataToCache(const char* url, const char* headerMetaData, const char* data, int size)
+     {
+       rtHttpCacheData cacheData(url,headerMetaData,data,size);
+       return rtFileCache::getInstance()->addToCache(cacheData);
+     }
+};
+
+class pxFileCacheTest : public testing::Test, public commonTestFns
+{
+  public:
+    virtual void SetUp()
+    {
+      
+    }
+  
+    virtual void TearDown()
+    {
+      rtFileCache::getInstance()->clearCache();
+      rtFileCache::destroy();      
+    }
+
+    void fileCacheSetMaxCacheSizeTest ()
+    { 
+       rtFileCache::getInstance()->setMaxCacheSize(200);
+       int64_t cacheSize;
+       cacheSize  = rtFileCache::getInstance()->maxCacheSize();
+       EXPECT_TRUE (cacheSize == 200);
+    }
+    
+    void fileCacheSetDirectoryTest()
+    { 
+      rtFileCache::getInstance()->setCacheDirectory("/tmp/cache");
+      rtString directory;
+      rtFileCache::getInstance()->cacheDirectory(directory);
+      EXPECT_TRUE (directory == "/tmp/cache");
+    }
+    
+    void fileCacheRemoveDataUrlNullTest()
+    { 
+      EXPECT_TRUE (rtFileCache::getInstance()->removeData(NULL) == RT_ERROR);
+    }
+    
+    void fileCacheRemoveDataUrlEmptyTest()
+    { 
+      EXPECT_TRUE (rtFileCache::getInstance()->removeData("") == RT_OK);
+    }
+    
+    void fileCacheRemoveDataUrlUnavailableTest()
+    { 
+      EXPECT_TRUE (rtFileCache::getInstance()->removeData("http://localhost/a.jpeg") == RT_OK);
+    }
+    
+    void fileCacheRemoveDataUrlAvailableTest()
+    { 
+      resetAndAddCacheData();
+      EXPECT_TRUE (rtFileCache::getInstance()->removeData("http://localhost/a.jpeg") == RT_OK); 
+      EXPECT_TRUE (rtFileCache::getInstance()->cacheSize() == 0); 
+    }
+    
+    void fileCacheRemoveDataSingleUrlAvailableTest()
+    { 
+      resetAndAddCacheData();
+      addDataToCache("http://localhost/b.jpeg","Expires: Sun 02 Oct 2016 22:33:33 UTC","abcde",5);
+      EXPECT_TRUE (rtFileCache::getInstance()->removeData("http://localhost/a.jpeg") == RT_OK); 
+      rtHttpCacheData data;
+      rtFileCache::getInstance()->getHttpCacheData("http://localhost/b.jpeg",data);
+      int expectedSize = strlen("Expires: Sun 02 Oct 2016 22:33:33 UTC") + 1 + strlen("abcde") + 1 + to_string(data.expirationDateUnix()).length();
```

Comment:
to_string() is not supported on all platforms.  Please see the previous note for suggestions.